core/modules/replica/AdminApp.cc

```diff
@@ -0,0 +1,296 @@
+/*
+ * LSST Data Management System
+ * Copyright 2018 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/AdminApp.h"
+
+// System headers
+#include <iomanip>
+#include <iostream>
+#include <stdexcept>
+#include <vector>
+
+// Qserv headers
+#include "replica/Controller.h"
+#include "replica/Performance.h"
+#include "replica/RequestTracker.h"
+#include "replica/ServiceManagementRequest.h"
+#include "util/TablePrinter.h"
+
+using namespace std;
+
+namespace {
+
+string const description {
```

Comment:
I would like to stay away from the row C-style strings as much as possible. I don't see any performance implications in using the C++ string class here.

Reply:
You cannot stay away from C-style strings because `"..."` is C-style string literal. Performance-wise your code is probably 5 million times less efficient than `const char[]` because it's global initialization and it has to be done even if you never use that variable.